### PR TITLE
ban + unban moderation & cmd & events test

### PR DESCRIPTION
**Pull request by ravlio**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: test/channels.js [hunk 1]
```diff
@@ -197,6 +197,41 @@ describe('Channels - members', function() {
 		await getTestClient(true).updateUser({ id: thierryID, role: 'admin' });
 	});
 
+	it('thierry bans tommaso', async function() {
+		let s;
+		const eventPromise = new Promise(resolve => {
+			s = tommasoClient.on('user.banned', e => {
+				expect(e.reason).to.equal('somereason');
+				expect(e.channel_id).to.equal(channelId);
```
Comment (tbarbugli) [inline test/channels.js]: please test channel_cid and channel_type as well

File: test/channels.js [hunk 2]
```diff
@@ -197,6 +197,41 @@ describe('Channels - members', function() {
...
+
+		await eventPromise;
+		s.unsubscribe();
+	});
+
+	it('thierry unbans tommaso', async function() {
+		let s;
+		const eventPromise = new Promise(resolve => {
+			s = tommasoClient.on('user.unbanned', () => {
+				resolve();
```
Comment (tbarbugli) [inline test/channels.js]: please add more checks on event data

File: test/serverside.js [hunk 3]
```diff
@@ -607,13 +607,84 @@ describe('Managing users', function() {
 	});
 
 	it('ban user', async function() {
+		const client1 = await getTestClientForUser(evilUser);
+		let s;
+		const eventPromise = new Promise(resolve => {
+			s = client1.on('user.banned', e => {
+				expect(e.reason).to.equal('reason');
```
Comment (tbarbugli) [inline test/serverside.js]: needs more event data tests (ie. no channel fields)

File: test/webhook.js [hunk 4]
```diff
@@ -423,6 +423,58 @@ describe('Webhooks', function() {
...
+		const text = `/ban ${tommasoID} reason`;
+		const [events] = await Promise.all([
+			promises.waitForEvents('user.banned'),
+			chan.sendMessage({ text, user_id: jaapID }),
+		]);
+		const event = events[0];
+		expect(event.reason).to.equal('reason');
+		expect(event.channel_id).to.equal(chan.id);
+		expect(event.channel_type).to.equal(chan.type);
+		expect(event.expiration).to.exist;
```
Comment (tbarbugli) [inline test/webhook.js]: you should verify that we return a datetime object that we can parse the same way as we do for created_at

File: test/serverside.js (hunk 3 above)
Comment (ravlio) [reply test/serverside.js]: Sorry missed this comment 🤦‍♂ 

there are no channel fields.

File: test/webhook.js (hunk 4 above)
Comment (ravlio) [reply test/webhook.js]: I have not found any datetime checks. The only I've found is:
`expect(userResponse.created_at).to.not.equal('0001-01-01T00:00:00Z')`

created_at is a string, not an object.

```
{
  type: 'user.banned',
  created_at: '2019-11-05T12:13:52.092106Z',
  reason: 'reason',
  expiration: '2019-11-05T14:16:52.091473Z',
  received_at: 2019-11-05T12:13:52.092Z
}
```

File: test/serverside.js (hunk 3 above)
Comment (Bogdan-D) [reply test/serverside.js]: probably we need to add channel fields? @ravlio

Comment (Bogdan-D) [reply test/serverside.js]: There are two kinds of `ban`, first is ban for channel, second is ban global for app. If user get's banned from channel there should be channel fields. In test you have to make sure there are no channel field for global ban, and there are channel fields for channel ban, I think. Fix me if I'm wrong

File: test/serverside.js [hunk 5]
```diff
@@ -607,13 +607,84 @@ describe('Managing users', function() {
...
+		const channel = client.channel('messaging', channelId, {
+			created_by: { id: createdById },
+		});
+		await channel.create();
+
+		const client1 = await getTestClientForUser(evilUser);
+		const eventPromise = new Promise(resolve => {
+			s = client1.on('user.banned', e => {
+				expect(e.reason).to.equal('reason');
+				expect(e.channel_id).to.equal(channelId);
```
Comment (Bogdan-D) [inline test/serverside.js]: channel type check is missing

**Pull request closed without merging**